src/pulsar_client.erl

```diff
@@ -383,10 +404,10 @@ do_try_connect([URI | Servers], Opts0, Res) ->
                     {ok, Result};
                 {error, Reason} ->
                     ok = close_socket_and_flush_signals(Sock, Opts),
-                    do_try_connect(Servers, Opts, Res#{{Host, Port} => Reason})
+                    {error, #{{Host, Port} => Reason}}
```

Comment:
it's now always a single host:port, maybe no need to return a map.
it's the caller's responsibility the make a map